frontend/src/pages/ProgressionChecker/TableView/TableView.jsx

```diff
@@ -1,53 +1,139 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { Table } from "antd";
+import { getMostRecentPastTerm } from "../../TermPlanner/validateTermPlanner";
 import "./index.less";
 
-const TableView = ({ checkercourses }) => {
+const TableView = ({
+  structure, years, startYear, courses,
+}) => {
+  const [pastCourses, setPastCourses] = useState({});
+  const [tableLayout, setTableLayout] = useState({});
+
+  const getPastCourses = () => {
+    const { Y: recentYear, T: recentTerm } = getMostRecentPastTerm(startYear);
+    const newPastCourses = {};
+
+    years.slice(0, recentYear).forEach((year, yearIndex) => {
+      Object.values(year).forEach((term, termIndex) => {
+        Object.values(term).forEach((course) => {
+          if (yearIndex < recentYear - 1 || termIndex <= recentTerm) {
+            const currCourse = {};
+            currCourse.title = courses[course].title;
+            currCourse.UOC = courses[course].UOC;
+            currCourse.type = courses[course].type;
+            const currYear = (startYear + yearIndex).toString().slice(-2);
+            currCourse.termTaken = `${currYear}T${termIndex}`;
+
+            newPastCourses[course] = currCourse;
+          }
+        });
+      });
+    });
+    setPastCourses(newPastCourses);
+  };
+
+  const generateTableStructure = (plannedCourses) => {
+    const newTableLayout = {};
+
+    // Example groups: Major, Minor, General
+    Object.keys(structure).forEach((group) => {
+      newTableLayout[group] = {};
+      // Example subgroup: Core Courses, Computing Electives, Flexible Education
+      Object.keys(structure[group]).forEach((subgroup) => {
+        if (typeof structure[group][subgroup] !== "string") {
+          // case where structure[group][subgroup] gives information on courses in an object
+          const subgroupStructure = structure[group][subgroup];
+          newTableLayout[group][subgroup] = [];
+
+          if (subgroupStructure.courses) {
+            // only consider disciplinary component courses
+            Object.keys(subgroupStructure.courses).forEach((courseCode) => {
+              if (courseCode in plannedCourses) {
+                newTableLayout[group][subgroup].push({
+                  key: courseCode,
+                  title: plannedCourses[courseCode].title,
+                  UOC: plannedCourses[courseCode].UOC,
+                  termTaken: plannedCourses[courseCode].termTaken,
+                });
+                newTableLayout[group][subgroup].sort(
+                  (a, b) => a.termTaken.localeCompare(b.termTaken),
+                );
+              }
+            });
+          } else {
+            // If there is no specified course list for the subgroup, then manually
+            // show the added courses on the menu.
```

Comment:
Change 'show the added courses on the menu' comment? Progression checker is not a menu? Lol

Reply:
oopsies 😆  yep will change